scripts/validate-test-results

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env perl
+
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use strict;
+use warnings;
+use experimental 'signatures';
+
+use Mojo::JSON qw(decode_json);
+use Mojo::Path;
+use Mojo::File qw(path);
+use Test::More;
+
+sub validate_test_result($result_file_path) {
+    my $result_file = path(Mojo::Path->new($result_file_path)->canonicalize);
+    my $test_result = decode_json($result_file->slurp)->{result};
+    return is $test_result, 'ok', sprintf ('Test result [%s] ok', $result_file->to_string);
+}
+
+my ($dir) = @ARGV;
+
+die "Missing argument DIR\n" unless $dir;
+
+my @results = glob("$dir/result-*.json");
+validate_test_result($_) for (@results);
+
+done_testing;
```

Comment:
For this file I think we stated in our last call(s) that multiple persons prefer either the former jq "hacky" way or an explicit option in isotovideo rather than a custom script in the example distribution which is a dangerous entry vector for yet another untested/unlinted code area. On top as you are making this a CLI application if we would follow "openQA standards" – which we should – then we would need to apply the rules from https://github.com/os-autoinst/os-autoinst/blob/master/t/44-scripts.t to here so e.g. making the command line parsing more helpful to unsuspecting users. I think it's a better sustainable approach to do that within isotovideo.

This does not mean that *you* in particular need to do that. This just means that before we have an implementation within isotovideo we should IMHO keep the current as-is approach, the jq "hacky" way.

Reply:
As stated in [poo#138416](https://progress.opensuse.org/issues/138416?next_issue_id=138365&prev_issue_id=138440#note-28) I'll do either of the two options:

  * provide the suggested `scripts/validate-test-results` from [os-autoinst/os-autoinst-distri-example#28][oade-28]
  * allow `isotoviodeo` to reflect the test result in the exit code. Something along the lines of `isotovideo [...] --exit-status-from-error`.

Whatever of them that constitute the less effort.

Reply:
The second option: os-autoinst/os-autoinst#2419

Reply:
Little hint regarding referencing github issues and pull requests. If you just reference https://github.com/os-autoinst/os-autoinst/pull/2419 inline then the link will be shortened but there is no preview of the title and status. If you use an entry in a list like in
* https://github.com/os-autoinst/os-autoinst/pull/2419

then it's providing more details.

Reply:
I see the same details when I hover over the link. It's a bit odd that it doesn't expand the same way, tho